test/extension/handlers/topologymutation/handler.go

```diff
@@ -19,70 +19,239 @@ package topologymutation
+	//    - kubeadmConfigSpec.InitConfiguration.KubeletExtraArgs
+	//    - kubeadmConfigSpec.JoinConfiguration.KubeletExtraArgs
+	_, cpVersion, err := getStringValueForVariable(templateVariables, "builtin.controlPlane.version")
+	if err != nil {
+		return errors.Wrap(err, "could not set cgroup-driver to control plane template kubeletExtraArgs")
+	}
+	controlPlaneVersion, err := version.ParseMajorMinorPatchTolerant(cpVersion)
 	if err != nil {
 		return err
 	}
+	if version.Compare(controlPlaneVersion, versionCeilingForPatch) == -1 {
+		log.Info(fmt.Sprintf("Setting kubeadmControlPlaneTemplate cgroup-driver %s", "cgroup-driver"))
+		if kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.InitConfiguration != nil {
+			kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.InitConfiguration.NodeRegistration.KubeletExtraArgs["cgroup-driver"] = cgroupDriverCgroupfs
+		}
+		if kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.JoinConfiguration != nil {
+			kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.JoinConfiguration.NodeRegistration.KubeletExtraArgs["cgroup-driver"] = cgroupDriverCgroupfs
+		}
+	}
 
-	// Unquote the JSON string.
-	stringValue, err := strconv.Unquote(string(value.Raw))
+	// 2) Patch RolloutStrategy RollingUpdate MaxSurge with the value from the Cluster Topology variables.
+	//    If this is unset continue as this variable is not required.
+	if found, kcpControlPlaneMaxSurge, err := getStringValueForVariable(templateVariables, "kubeadmControlPlaneMaxSurge"); found {
+		if err != nil {
+			return errors.Wrap(err, "could not set KubeadmControlPlaneTemplate MaxSurge")
+		}
+		// This has to be converted to IntOrString type.
+		kubeadmControlPlaneMaxSurgeIntOrString := intstrutil.Parse(kcpControlPlaneMaxSurge)
+		log.Info(fmt.Sprintf("Setting kubeadmControlPlaneMaxSurge %s", kubeadmControlPlaneMaxSurgeIntOrString.String()))
+		if kubeadmControlPlaneTemplate.Spec.Template.Spec.RolloutStrategy == nil {
+			kubeadmControlPlaneTemplate.Spec.Template.Spec.RolloutStrategy = &controlplanev1.RolloutStrategy{}
+		}
+		if kubeadmControlPlaneTemplate.Spec.Template.Spec.RolloutStrategy.RollingUpdate == nil {
+			kubeadmControlPlaneTemplate.Spec.Template.Spec.RolloutStrategy.RollingUpdate = &controlplanev1.RollingUpdate{}
+		}
+		kubeadmControlPlaneTemplate.Spec.Template.Spec.RolloutStrategy.RollingUpdate.MaxSurge = &kubeadmControlPlaneMaxSurgeIntOrString
+	}
+	return nil
+}
+
+// patchDockerClusterTemplate patches the DockerClusterTemplate.
+func patchDockerClusterTemplate(_ context.Context, dockerClusterTemplate *infrav1.DockerClusterTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
+	_, lbImageRepo, err := getStringValueForVariable(templateVariables, "lbImageRepository")
 	if err != nil {
-		return err
+		return errors.Wrap(err, "could not set DockerClusterTemplate loadBalancer imageRepository")
+	}
```

Comment:
I would prefer also doing this only if the variable is found

Reply:
Looking at the `getStringValueForVariable` this should work given that if we don't find the variable we return an empty string and setting something to the empty string is a no-op.

But I think it's way more explicit and thus easier to read if we explicitly handle found

Same for other usages of `getStringValueForVariable` below

Reply:
Same for `patchKubeadmConfigTemplate`. We can use the `if !found` code there to add an explanation in godoc that in this case the KubeadmConfigTemplate does not belong to any MachineDeployment